### PR TITLE
Partially implements #127

### PR DESCRIPTION
**Pull request by Jaifroid**
This implements the most urgent aspects of #127 (Reopen the last article after closing/reopening, or after switching between menu items). It implements a link to go back to the article, and disables the link if a major UI option is changed (so as to force a reload). It does not implement storing the article title in a cookie.

Please note that this PR also contains a channel for posting a message to the user on the "spinner" page, which is available here for future use (I use it on kiwix-js-windows). As these changes are very closely related, it seemed best to include it as an available utility in uiUtil.js. Messages posted to the user via uiUtil.poll(msg) will only be visible if the interface is taking a long time to load a file, hence only really on mobile with heavy pages.

### CODE REVIEW

**Comment by mossroy** [comment]
One minor comment : you don't have to put the "implements #xxx" in the title of your commits or pull requests. It would be more readable to put a meaningful title, and put this string below (in the following lines of the commit comment), or in the PR description. GitHub will use them to close the corresponding issue too.

File: www/index.html [hunk 1]
```diff
@@ -189,6 +189,7 @@
                 </div>
 
                 <div id="configuration" style="display: none;" class="category-padding">
+                <div id="returntoArticle_top" style="text-align: center;"></div>
```
Comment (mossroy) [inline www/index.html]: Instead of adding specific links, I would find more user-friendly to return to the previous article when clicking on the Home button (instead of reloading the home page).
It would also work when coming back from the About section.

File: www/js/app.js [hunk 2]
```diff
@@ -771,15 +773,19 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
      * @param {String} htmlArticle
      */
     function displayArticleInForm(dirEntry, htmlArticle) {
+        // Display the article inside the web page.
+
+        //Fast-replace img with data-img and hide image [kiwix-js #272]
+        htmlArticle = htmlArticle.replace(/(<img\s+[^>]*\b)src(\s*=)/ig, "$1data-kiwixsrc$2");
```
Comment (mossroy) [inline www/js/app.js]: This does not seem to be related to #127. I suppose it's because you did not create your branch from master?

File: www/js/lib/uiUtil.js [hunk 3]
```diff
@@ -53,11 +53,33 @@ define([], function() {
...
+        var link = '<h4><a href="#" onclick="$(\'#configuration\').hide();$(\'#formArticleSearch\').show();' +
+            '$(\'#articleContent\').show();$(\'#liHomeNav\').attr(\'class\',\'active\');$(\'#liConfigureNav\').attr(\'class\', \'\');' +
+            '$(\'#liAboutNav\').attr(\'class\', \'\');">&lt;&lt; Return to ' + shortTitle + '</a></h4>';
+        document.getElementById("returntoArticle_top").innerHTML = link;
+        document.getElementById("returntoArticle_bottom").innerHTML = link;
+    }
+
+    function poll(msg) {
+        document.getElementById("progressMessage").innerHTML += '<p>' + msg + '</p>';
+    }
```
Comment (mossroy) [inline www/js/lib/uiUtil.js]: I don't see this function ever called?

File: www/js/lib/uiUtil.js [hunk 4]
```diff
@@ -53,11 +53,33 @@ define([], function() {
         }
     }
 
+    function makeReturnLink(title) {
+        //Abbreviate title if necessary
+        var shortTitle = title.substring(0, 25);
+        shortTitle = shortTitle == title ? shortTitle : shortTitle + "..."; 
+        var link = '<h4><a href="#" onclick="$(\'#configuration\').hide();$(\'#formArticleSearch\').show();' +
+            '$(\'#articleContent\').show();$(\'#liHomeNav\').attr(\'class\',\'active\');$(\'#liConfigureNav\').attr(\'class\', \'\');' +
+            '$(\'#liAboutNav\').attr(\'class\', \'\');">&lt;&lt; Return to ' + shortTitle + '</a></h4>';
```
Comment (mossroy) [inline www/js/lib/uiUtil.js]: I think we should avoid putting some HTML inside javascript.
This HTML code could probably be put in the HTML file, with the dynamic part identified by an id and filled from the javascript. Plus using the visibility to show/hide it.

File: www/index.html (hunk 1 above)
Comment (Jaifroid) [inline www/index.html]: I agree. This was more a quick solution attempt, especially given the discoverability issues on mobile (home button is hidden inside a clunky hamburger menu, taking more clicks to open and navigate around). Hence attempt (not very elegant, I agree!) to provide clear visual clues.

File: www/js/app.js (hunk 2 above)
Comment (Jaifroid) [inline www/js/app.js]: I just moved that code, it's not an addition. I did create the branch from master, and backported code into it from kiwix-js-windows. This move was "en passant".

File: www/js/lib/uiUtil.js (hunk 4 above)
Comment (Jaifroid) [inline www/js/lib/uiUtil.js]: I tried, but having "<h4>" in the HTML caused an obvious gap, even when empty. I would have had to have added "element.hide()" code in several different places I was uncertain of.... It's definitely a hack...

File: www/js/lib/uiUtil.js (hunk 3 above)
Comment (Jaifroid) [inline www/js/lib/uiUtil.js]: Indeed, it's for future use as mentioned in the PR comment -- a message channel.

**Comment by Jaifroid** [comment]
Thanks for the comments, @mossroy , I appreciate your guidance as I haven't grasped fully the intricacies of the app.

This was admittedly a quick attempt to add a feature that has been in the works for quite a while and is a real usability issue with the app,, as our Ubuntu Phone friend pointed out. There's also a discoverability issue on a phone: you have to open the hamburger menu in order to navigate between the tabs, so in-app navigation isn't intuitive on a phone.

I feel a few things need fixing urgently before I can put kiwix-js-windows in the Store, but I don't have very much time, hence the rush -- and apologies for that. It can certainly be done better than this, though I don't think I have the knowledge of the libraries to do things in a standard way. I'm happy to close the PR and leave the branch for a while in case anyone else can have a go at doing it properly.

**Comment by mossroy** [comment]
Don't apologize, it's still a step towards a fix.

No special library is used to switch between sections : it's simply programmatically hiding/un-hiding some parts of the DOM when the user clicks on some menu items.
At some point, the article content is emptied : this should be done at another moment, and the home page should be displayed only if the content is empty
